src/Mvc/Mvc.Core/src/UrlHelperExtensions.cs

```diff
@@ -226,6 +226,16 @@ public static class UrlHelperExtensions
         });
     }
 
+    private static string TrimAsyncSuffix(string actionName)
+    {
+        const string suffix = "Async";
+        if (actionName.EndsWith(suffix, false, CultureInfo.InvariantCulture))
```

Comment:
Doing this unconditionally wouldn't be right (if say the option to trim the Async suffix is disabled). Could you describe a possible solution in the issue before we start writing code? Thanks!